### PR TITLE
feat: streamline metrics

### PR DESCRIPTION
**Pull request by zoumo**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind feature

**What this PR does / why we need it**:
feat: streamline metrics

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/gateway/proxy/dispatcher/proxylog.go [hunk 1]
```diff
@@ -95,6 +98,10 @@ func (rw *responseWriterDelegator) Write(b []byte) (int, error) {
 		rw.debugf("logging error output: %q\n", string(b))
 	}
 	n, err := rw.w.Write(b)
+	if rw.isWatch() {
+		// do not block write
+		go metrics.RecordWatchEventSize(rw.host, rw.endpoint, rw.requestInfo.Resource, n)
```
Comment (xuchen-xiaoying) [inline pkg/gateway/proxy/dispatcher/proxylog.go:103]: every watch event will create a new go routine here?

Comment (zoumo) [reply pkg/gateway/proxy/dispatcher/proxylog.go]: solved

**Review by xuchen-xiaoying — APPROVED** [review]
LGTM

**Review by xuchen-xiaoying — APPROVED** [review]
LGTM

**Pull request merged**